### ISSUE TITLE
azurerm_lighthouse_definition undocumented "scope" argument is required

### ISSUE THREAD

**Issue by lukiffer** [terraform-providers/terraform-provider-azurerm#10120]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform 0.14.4
AzureRM Provider Version 2.42.0

### Affected Resource(s)

* `azurerm_lighthouse_definition`

### Terraform Configuration Files

```hcl
resource "azurerm_lighthouse_definition" "definition" {
  name               = "Sample definition"
  description        = "This is a lighthouse definition created via Terraform"
  managing_tenant_id = "00000000-0000-0000-0000-000000000000"

  authorization {
    principal_id       = "00000000-0000-0000-0000-000000000000"
    role_definition_id = "00000000-0000-0000-0000-000000000000"
  }
}
```

### Expected Behaviour

This is effectively a duplicate of #8955. This issue was closed, but the issue remains that the validation step fails stating that an [undocumented](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lighthouse_definition#argument-reference) argument `scope` is required.

Moreover, this resource doesn't describe a `scope`, but the `azurerm_lighthouse_assignment` does.

Either the validation is incorrect, or the docs should be updated to list this argument.

The resource should be provisioned.

### Actual Behaviour

A validation error is thrown:

```
Error: Missing required argument

  on main.tf line 5, in resource "azurerm_lighthouse_definition" "definition":
   5: resource "azurerm_lighthouse_definition" "definition" {

The argument "scope" is required, but no definition was found.
```

### Steps to Reproduce

1. `terraform apply`

### References

* #8955

**Comment by lukiffer** [comment]
The argument is defined in the schema:
https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/lighthouse/lighthouse_definition_resource.go#L52

And it _is_ used:
https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/lighthouse/lighthouse_definition_resource.go#L116

It's not clear though what its purpose is, given that a definition can have multiple assignments, each with their own scope (separate resource `azurerm_lighthouse_assignment`). Is this merely the ID of the subscription in which it's being provisioned or is it something else? With this being unclear, would need an SME to provide clarity for documentation.

**Comment by lukecalderon** [comment]
Hit the same issue today with this...

Quick workaround until it's resolved...
`
resource "azurerm_lighthouse_definition" "reader" {
  name                 = "Global-Reader"
  description          = "Subscription-level reader role assignment to MSP"
  managing_tenant_id   = local.msp_tenant_id
  **scope                = format("/subscriptions/%s", var.subscription_id)**

  authorization {
    principal_id       = local.msp_principal_id
    role_definition_id = data.azurerm_role_definition.reader.role_definition_id
  }
}
`

**Comment by njuCZ** [comment]
It has been documented here: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lighthouse_definition#scope

**Comment by lukiffer** [comment]
This was fixed in #10997.
